witchcraft/witchcraft.go

```diff
@@ -440,6 +444,12 @@ func (s *Server) WithDisableKeepAlives() *Server {
 	return s
 }
 
+// WithStrictUnmarshalConfig configures the server to use the provided strict unmarshal configuration.
+func (s *Server) WithStrictUnmarshalConfig(strictUnmarshalConfig bool) *Server {
```

Comment:
can we make this be two methods `WithStrictUnmarshalConfig` and `WithoutStrictUnmarshalConfig` instead and no params? slight preference for APIs that don't involve boolean parameters.

Reply:
Sure -- for now, just making this `WithStrictUnmarshalConfig`. If we change default to strict, we can introduce the `Without` variant.